### PR TITLE
Re-design ssh Key as Verifier only

### PR DESCRIPTION
**Pull request by lukpueh**
Previously, ssh `Key` satisfied both the `dsse.Verifier` interface and served as TUF metadata key container. Unfortunately, it didn't seem feasible to wire up the key container with the current TUF metadata implementation, which uses `SSlibKey`.

This commit re-designs the ssh key implementation to use SSlibKey as key container and a separate Verifier for verification.

See https://github.com/gittuf/gittuf/pull/429#issuecomment-2151588628 for more detailed design considerations.

Change details:

* Add an SSHKeyTypeConstant
* Rename Import to NewKeyFromFile
* Replace ssh Key with ssh Verifier, which satisfies the dsse.Verifier interface, but is otherwise a black box (no key details exposed).
* Change NewKeyFromFile to return an SSlibKey, to be included in TUF metadata.
* Add NewVerifierFromKey function to create an ssh Verifier from a  corresponding SSlibKey, e.g. included in TUF metadata, to verify  a signature from an ssh Signer.
* Add TUF metadata integration POC / usage example in tuf_test.go (see [comment in #429](https://github.com/gittuf/gittuf/pull/429#discussion_r1628103131)).

### CODE REVIEW

**Comment by lukpueh** [comment]
Note that unlike #429, this PR does not add a "ssh Verifier to SSlibKey" conversion function, because with the re-design there does not seem to be a use case.

The general workflow should be:

1. Import SSlibKey from ssh key file
2. Add to TUF metadata, or cache otherwise
3. Convert Key to Verifier when verifying a related signature

**Comment by lukpueh** [comment]
Btw. creating a Signer is currently a bit cumbersome:

https://github.com/gittuf/gittuf/blob/bd5d1d715414e66910e00fa1bc0d23e642af60cb/internal/signerverifier/dsse/dsse_test.go#L65-L78

`NewSignerFromFile(path string)` or `NewSignerFromKey(key SSLibKey, path string)` functions would be helpful. I can add those in a separate PR, if there is a need.

**Review by adityasaky — APPROVED** [review]

File: internal/signerverifier/ssh/ssh.go [hunk 1]
```diff
@@ -45,7 +37,7 @@ func (k *Key) Verify(_ context.Context, data []byte, sig []byte) error {
 
 	// ssh-keygen uses sha512 to sign with **any*** key
 	hash := sshsig.HashSHA512
-	if err = sshsig.Verify(message, signature, pub, hash, SSHSigNamespace); err != nil {
+	if err := sshsig.Verify(message, signature, v.sshKey, hash, SSHSigNamespace); err != nil {
```
[the remark this answers is not in the document]
Comment (lukpueh) [reply internal/signerverifier/ssh/ssh.go:40]: Good question. Both work. You can see the difference here: https://go.dev/play/p/0F5yN3WQmhK

I have no idea, which ones is recommended.

[the remark this answers is not in the document]
Comment (adityasaky) [reply internal/signerverifier/ssh/ssh.go:40]: I usually keep err scoped to the if block when used this way, so with the `:`. I think it doesn't matter either way, most of the time, though.

**Pull request merged**